### PR TITLE
ARROW-4358: [CI] Restore support for trusty in CI

### PR DESCRIPTION
**Pull request by pravindra**
- also, add travis entry for trusty

### CODE REVIEW

File: ci/travis_install_linux.sh [hunk 1]
```diff
@@ -37,13 +37,13 @@ fi
...
 
-sudo apt-get install -y -qq maven
+#sudo apt-get install -y -qq maven
 
 # Remove Travis-specific versions of Java
-sudo rm -rf /usr/local/lib/jvm*
-sudo rm -rf /usr/local/maven*
-hash -r
-unset JAVA_HOME
+#sudo rm -rf /usr/local/lib/jvm*
```
Comment (praveenbingo) [inline ci/travis_install_linux.sh]: should we instead add a flag to remove this for Xenial and not for trusty..

Comment (praveenbingo) [reply ci/travis_install_linux.sh]: also should we try out a dremio/arrow-build with this patch to catch any other divergences..

**Comment by pravindra** [comment]
I'm hitting this error in CI for trusty. Any idea what could be wrong ?

/usr/bin/ld: /home/travis/build/apache/arrow/cpp-toolchain/lib/libdouble-conversion.a(double-conversion.cc.o): unrecognized relocation (0x29) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status

**Comment by wesm** [comment]
Haha, this is what I was talking about in https://issues.apache.org/jira/browse/ARROW-4343

To use the conda-forge packages on Trusty you have to install some additional packages: `gcc_linux-64` and `gxx_linux-64`. gcc-4.8 can no longer be used together with the conda-forge packages

**Comment by pravindra** [comment]
@wesm, can you please review?

Installing the gcc and gxx packages from Vonda-forge didn’t help - there were tonnes of linking errors. Maybe, it needs to be accompanied by some other step that I missed.

Setting the conda forge label got the build working in both Travis CI and cross-bow (it also uses the same CI scripts).

File: ci/travis_install_linux.sh (hunk 1 above)
Comment (pravindra) [reply ci/travis_install_linux.sh]: Done for both.

**Comment by wesm** [comment]
I'll try to look this weekend. In exchange if you could help by reviewing some other PRs it would be much appreciated

**Comment by wesm** [comment]
Those conda packages only work if you install them into the environment before calling `conda activate`. there is a special script that sets a bunch of environment variables

**Comment by wesm** [comment]
Oh actually I forgot that on Trusty there are problems with pthread

I started working on that in https://issues.apache.org/jira/browse/ARROW-4330. I'll try to get that fixed tomorrow and then see if that fixes your build

**Comment by wesm** [comment]
Sorry that I haven't got to this yet. I can look sometime tomorrow but it may not be until evening US/Eastern time

**Comment by pravindra** [comment]
If it’s okay, can you please merge my current change (CI succeeds), and improve on it in another PR ?

**Comment by wesm** [comment]
OK, no problem. I'll look at it more later

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**